apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnection.cs

```diff
@@ -874,5 +882,31 @@ private void EnlistTransaction(Transaction transaction, TimestampBound timestamp
 
         /// <inheritdoc />
         protected override DbProviderFactory DbProviderFactory => SpannerProviderFactory.Instance;
+
+        /// <summary>
+        /// Get Resource based routing flag from environment variable.
+        /// </summary>
+        /// <returns>resource based routing flage</returns>
+        private static bool GetResourceBasedRoutingFlag()
+        {
+            string resourceBaseRoutingFlagValue =
+                Environment.GetEnvironmentVariable("GOOGLE_CLOUD_ENABLE_RESOURCE_BASED_ROUTING")?.ToLower();
+            return !string.IsNullOrEmpty(resourceBaseRoutingFlagValue) &&
+                resourceBaseRoutingFlagValue.Equals("true");
+        }
+
+        /// <summary>
+        /// Override the default endpointUri with available endpointUri.
+        /// </summary>
+        /// <param name="endpointUri"></param>
+        private void OverrideSpannerClientEndpoints(string endpointUri)
+        {
+            if (!string.IsNullOrEmpty(endpointUri))
```

Comment:
Do you want to put this condition out and only call OverrideSpannerClientEndpoints on `true` (on line 647)?